core/src/main/java/org/infinispan/persistence/PersistenceUtil.java

```diff
@@ -188,4 +190,30 @@ public static int count(SegmentedAdvancedLoadWriteStore<?, ?> salws, IntSet segm
       return org.infinispan.persistence.internal.PersistenceUtil.parallelizePublisher(segments, Schedulers.from(executor),
             publisherFunction);
    }
+
+   /**
+    * Replace unwanted characters from cache names so they can be used as filenames
+    */
+   public static String sanitizedCacheName(String cacheName) {
+      return cacheName.replaceAll("[^a-zA-Z0-9-_\\.]", "_");
+   }
+
+   public static Path getQualifiedLocation(GlobalConfiguration globalConfiguration, String location, String cacheName, String qualifier) {
+      Path persistentLocation = Paths.get(globalConfiguration.globalState().persistentLocation());
+      if (location == null) {
+         return persistentLocation.resolve(Paths.get(sanitizedCacheName(cacheName), qualifier));
+      } else {
+         Path path = Paths.get(location);
+         if (path.isAbsolute()) {
```

Comment:
I think the point of an absolute path is exactly that, to be able to get out of the 'cage'. Can't we trust users knowing what they do?

Reply:
But in managed environments you don't want to allow users creating caches wherever

Reply:
For flexibility, I can add a "strict persistent path" boolean attribute to global state which defaults to true, but which can be turned off.

Reply:
I'm more against having an additional switch than the restriction tbh, so if we're going todo it lets make it mandatory.